### PR TITLE
Add IEEE formatter

### PR DESCRIPTION
**Pull request by kk6axq**
This adds a basic IEEE formatter. It formats dates as "YYYY, [Word month] day" and formats authors as "[first initial] [last name]" for each author, or [first author initial and last name] et al" if >6 authors, per [reference guide].(https://comunidad.udistrital.edu.co/redesdeingenieria/files/2013/04/Referencias-IEEE.pdf)

It also adds `ieee` as a valid option to the `--format` command lie argument

### CODE REVIEW

File: src/python_autocite/formatter/ieee.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            if (len(first_last) > 1):
+                # This pulls the last name and first initial
+                formatted_names.add(first_last[0][0] + ", " + first_last[1])
+
+        authors = list(formatted_names)
+        if (len(authors) < 1):
+            return self.AUTHOR_UNKNOWN
+
+        authors.sort()
+        if (len(authors > 6)):
```
Comment (thenaterhood) [inline src/python_autocite/formatter/ieee.py]: This line has a misplaced parenthesis.

`if (len(authors > 6)):` <-- current code
`if (len(authors) > 6):` <-- the needed correction

**Review by thenaterhood — CHANGES_REQUESTED** [review]
Thanks for the contribution! I left one comment on a problem that needs to be fixed, but in general this looks good.

File: src/python_autocite/formatter/ieee.py (hunk 1 above)
Comment (kk6axq) [reply src/python_autocite/formatter/ieee.py]: Thanks for catching that. [ce2202d](https://github.com/thenaterhood/python-autocite/pull/8/commits/ce2202d118854733ea972799510288ab21b1eb41) fixes that.

**Comment by kk6axq** [comment]
That problem is fixed now

**Comment by thenaterhood** [comment]
Looks good! Thank you very much :). Merged.

**Pull request merged**